src/modules/train_controller/inc/train_controller.h

```diff
@@ -6,4 +6,193 @@
+            {16, {50.0, 0.0, 50.0, 0.00, 0.00}},     // added 16th block
+        };
+
+        // map for block data -->
+        // key:
+        //          block number
+        // values:
+        //          infrastructure string
+
+        std::unordered_map<types::BlockId, std::string> blueBlockInfrastructureMap = {
+            {3, "RAILWAY CROSSING"},
+            {5, "Switch ( 5 to 6) or (5 to 11)"},
+            {6, "Switch ( 5 to 6); Light"},
+            {9, "Transponder"},
+            {10, "Station B"},
+            {11, "Switch (5 to 11); Light"},
+            {14, "Transponder"},
+            {15, "Station C"},
+        };
+
+    public:
+        // Implementations for getters
+        types::MilesPerHour GetCurrentVelocity(void) const;
+        types::MilesPerHour GetDriverSpeed(void) const;
+        types::MilesPerHour GetCommandedSpeedWS(void) const;
+        types::Meters GetDistanceTravelled(void) const;
+        types::Watts GetCommandedPower(void) const;
+        uint16_t GetServiceBrake(void) const;
+        uint16_t GetGrade(void) const;
+        bool GetEmergencyBrake(void) const;
+        bool GetHeadLights(void) const;
+        bool GetInteriorLights(void) const;
+        bool GetLeftDoors(void) const;
+        bool GetRightDoors(void) const;
+        bool GetBrakeFailure(void) const;
+        bool GetSignalPickupFailure(void) const;
+        bool GetEngineFailure(void) const;
+        types::DegreesFahrenheit GetInternalTemperature(void) const;
+        types::DegreesFahrenheit GetActInternalTemperature(void) const;
+        types::Meters GetAuthority(void) const;          // Getter for Authority
+
+        // Implementations for setters
+        void SetCommandedSpeedWS(types::MilesPerHour speed);
+        void SetDriverSpeed(types::MilesPerHour speed);
+        void SetCurrentVelocity(types::MilesPerHour velocity);
+        void SetServiceBrake(uint16_t percentage);
+        void setEmergencyBrake(bool state);
+        void SetHeadLights(bool state);
+        void SetInteriorLights(bool state);
+        void SetLeftDoors(bool state);
+        void SetRightDoors(bool state);
+        void SetBrakeFailure(bool state);
+        void SetSignalPickupFailure(bool state);
+        void SetEngineFailure(bool state);
+        void SetInternalTemperature(types::DegreesFahrenheit temp);
+        void SetActInternalTemperature(types::DegreesFahrenheit temp);
+        void SetAuthority(types::Meters authority);
+        void SetArrived(bool);
+        void SetKP(uint16_t);
+        void setKI(uint16_t);
+
+        //TestBench Functions. These will be tied to the testbench UI
+        virtual void SetCommandedSpeedTB(types::MilesPerHour);
+        virtual void SetCurrentVelocityTB(types::MilesPerHour);
+
+        //local functions
+        virtual void CalculateCommandedPower();
+        virtual void UpdateDistanceTravelled(long interval);
+        void Update();
+
+    private:
```

Comment:
make sure all member variables follow the naming conventions here: https://bte12.atlassian.net/wiki/spaces/NN/pages/1081369/C+Coding+Standard#Variable-Names,

and initialize members in the constructor